packages/database/src/core/view/QueryParams.ts

```diff
@@ -194,211 +198,237 @@ export class QueryParams {
+
+  if (queryParams.isDefault()) {
+    return qs;
   }
 
-  isDefault(): boolean {
-    return this.loadsAllData() && this.index_ === PRIORITY_INDEX;
+  let orderBy;
+  if (queryParams.index_ === PRIORITY_INDEX) {
+    orderBy = REST_QUERY_CONSTANTS.PRIORITY_INDEX;
+  } else if (queryParams.index_ === VALUE_INDEX) {
+    orderBy = REST_QUERY_CONSTANTS.VALUE_INDEX;
+  } else if (queryParams.index_ === KEY_INDEX) {
+    orderBy = REST_QUERY_CONSTANTS.KEY_INDEX;
+  } else {
+    assert(queryParams.index_ instanceof PathIndex, 'Unrecognized index type!');
+    orderBy = queryParams.index_.toString();
   }
+  qs[REST_QUERY_CONSTANTS.ORDER_BY] = stringify(orderBy);
 
-  getNodeFilter(): NodeFilter {
-    if (this.loadsAllData()) {
-      return new IndexedFilter(this.getIndex());
-    } else if (this.hasLimit()) {
-      return new LimitedFilter(this);
-    } else {
-      return new RangedFilter(this);
+  if (queryParams.startSet_) {
+    qs[REST_QUERY_CONSTANTS.START_AT] = stringify(queryParams.indexStartValue_);
+    if (queryParams.startNameSet_) {
+      qs[REST_QUERY_CONSTANTS.START_AT] +=
+        ',' + stringify(queryParams.indexStartName_);
     }
   }
 
-  /**
-   * Returns a set of REST query string parameters representing this query.
-   *
-   * @return query string parameters
-   */
-  toRestQueryStringParameters(): { [k: string]: string | number } {
-    const REST_CONSTANTS = QueryParams.REST_QUERY_CONSTANTS_;
-    const qs: { [k: string]: string | number } = {};
-
-    if (this.isDefault()) {
-      return qs;
+  if (queryParams.endSet_) {
+    qs[REST_QUERY_CONSTANTS.END_AT] = stringify(queryParams.indexEndValue_);
+    if (queryParams.endNameSet_) {
+      qs[REST_QUERY_CONSTANTS.END_AT] +=
+        ',' + stringify(queryParams.indexEndName_);
     }
+  }
 
-    let orderBy;
-    if (this.index_ === PRIORITY_INDEX) {
-      orderBy = REST_CONSTANTS.PRIORITY_INDEX;
-    } else if (this.index_ === VALUE_INDEX) {
-      orderBy = REST_CONSTANTS.VALUE_INDEX;
-    } else if (this.index_ === KEY_INDEX) {
-      orderBy = REST_CONSTANTS.KEY_INDEX;
+  if (queryParams.limitSet_) {
+    if (queryParams.isViewFromLeft()) {
+      qs[REST_QUERY_CONSTANTS.LIMIT_TO_FIRST] = queryParams.limit_;
     } else {
-      assert(this.index_ instanceof PathIndex, 'Unrecognized index type!');
-      orderBy = this.index_.toString();
+      qs[REST_QUERY_CONSTANTS.LIMIT_TO_LAST] = queryParams.limit_;
     }
-    qs[REST_CONSTANTS.ORDER_BY] = stringify(orderBy);
+  }
 
-    if (this.startSet_) {
-      qs[REST_CONSTANTS.START_AT] = stringify(this.indexStartValue_);
-      if (this.startNameSet_) {
-        qs[REST_CONSTANTS.START_AT] += ',' + stringify(this.indexStartName_);
-      }
-    }
+  return qs;
+}
 
-    if (this.endSet_) {
-      qs[REST_CONSTANTS.END_AT] = stringify(this.indexEndValue_);
-      if (this.endNameSet_) {
-        qs[REST_CONSTANTS.END_AT] += ',' + stringify(this.indexEndName_);
-      }
+export function queryParamsGetQueryObject(
```

Comment:
FYI It looks like you re-ordered some of the methods, which makes this a internals impossible to review. I trust that you updated the methods correctly.

Reply:
Tests are passing!